journald/journald.cpp

```diff
@@ -100,11 +129,22 @@ class JournaldLoggerProcess : public Process<JournaldLoggerProcess>
           return Nothing();
         }
 
-        // Write the bytes to journald.
-        Try<Nothing> result = write(readSize);
-        if (result.isError()) {
-          promise.fail("Failed to write: " + result.error());
-          return Nothing();
+        if (flags.destination_type != "sandbox") {
```

Comment:
`if (flags.destination_type == "journald" || flags.destination_type == "sandbox_and_journald") {`
Or maybe `flags.destination_types` should take a list of strategies to follow, and you just have to check if the list contains "journald" or "sandbox"